src/test/util/event.rs

```diff
@@ -143,6 +143,55 @@ impl EventClient {
         }
     }
 
+    pub async fn merge_options(
```

Comment:
I have a moderate preference for calling this something like `with_additional_options`; `merge_options` doesn't quite sound like the name of a constructor to me, since `merge` sounds like verb that's just used to combine options, not create something else entirely